apps/dashboard/app/page.tsx

```diff
@@ -4,9 +4,130 @@ import { authOptions } from "./api/auth/[...nextauth]/route"
 
 import PriceContainer from "@/components/price-container/price-container"
 import ContentContainer from "@/components/content-container"
+import { TransactionEdge } from "@/services/graphql/generated"
+import {
+  fetchFirstTransactions,
+  fetchPaginatedTransactions,
+} from "@/services/graphql/queries/get-transactions"
+
+export async function fetchAllTransactions({
+  token,
+  days,
+}: {
+  token: string
+  days: number
+}) {
+  const allTransactions = []
+
+  // fetch all transactions for last n days
+  const daysAgo = Math.floor(Date.now() / 1000) - days * 24 * 60 * 60
+
+  let response = await fetchFirstTransactions(token)
+  let transactions = response?.edges || []
+  let pageInfo = response?.pageInfo
+
+  allTransactions.push(...transactions)
+
+  let lastTransactionDate =
+    transactions.length > 0 ? transactions[transactions.length - 1].node.createdAt : 0
+
+  while (pageInfo?.hasNextPage && lastTransactionDate > daysAgo) {
+    const nextCursor = pageInfo.endCursor
+    response = await fetchPaginatedTransactions(token, "next", nextCursor)
+    transactions = response?.edges || []
+    pageInfo = response?.pageInfo
+    allTransactions.push(...transactions)
+
+    lastTransactionDate =
+      transactions.length > 0 ? transactions[transactions.length - 1].node.createdAt : 0
+  }
+
+  return allTransactions
+}
```

Comment:
let's create a separate "library" or folder to make those calculation and add a batteries of unit test on some mock data to verify those are corrects